### ISSUE TITLE
spurious unused-import with typing.BinaryIO and type comments

### ISSUE THREAD

**Issue by chadrik** [PyCQA/pylint#3217]
### Steps to reproduce

Create this file:

```python
from typing import BinaryIO
from typing import List

def foo():
    # type: () -> BinaryIO
    pass

def foo():
    # type: () -> List
    pass
```

### Current behavior

```
pylint pylint_test.py 
************* Module pylint_test
pylint_test.py:1:0: W0611: Unused BinaryIO imported from typing (unused-import)

------------------------------------------------------------------
Your code has been rated at 5.00/10 (previous run: 6.67/10, -1.67)
```

### Expected behavior

There should not be an error because BinaryIO is used.  

It's interesting to note that the behavior is correct for List.

Also, the behavior is correct (no errors) using non-comment annotations

```python
from typing import BinaryIO
from typing import List

def foo() -> BinaryIO:
    pass

def foo() -> List:
    pass
```

### pylint --version output

```
pylint 2.4.3
astroid 2.3.1
Python 3.7.3 (default, Jun 19 2019, 07:40:15) 
[Clang 10.0.0 (clang-1000.11.45.5)]
```
